agent_shared/src/url_utils.rs

```diff
@@ -0,0 +1,120 @@
+pub trait UrlAddFunctions {
+    fn add_path(&self, path: &str) -> url::Url;
+    fn add_file(&self, file: &str) -> url::Url;
+}
+
+fn create_trailing_slash_url(url: &url::Url) -> url::Url {
+         if !url.path().ends_with('/') {
+             let res = url::Url::parse(&format!("{}/", url)).unwrap();
+             tracing::info!("res: {:?}", res);
+             res
+        } else {
+            url.clone()
+        }
+}
+
+impl UrlAddFunctions for url::Url {
+    fn add_path(&self, path: &str) -> url::Url {
+        let mut path = path.to_string();
+
+        if path.starts_with('/') {
+            path.remove(0);
+        }
+
+        if !path.ends_with('/') {
+            path.push('/')
+        }
+
+        let url = create_trailing_slash_url(self).join(&path);
+
+        println!("{:?}", url);
+
+        match url {
+            Ok(url) => url,
+            Err(err) => {
+                let err_str = format!("Path can't be added: {:?}\n{:?}", path, err);
+                tracing::error!("{:?}", &err_str);
+                panic!("{:?}", &err_str);
+            }
+        }
+    }
+
+    fn add_file(&self, file: &str) -> url::Url {
```

Comment:
I think this function name is a little bit misleading, `add_file()` sounds to me like I would actually supply some some bytes to create an actual file. Maybe we could rename to `append_file_name()` or something alike?